src/app/components/EditableText.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from 'react';
+
+import edit from '@assets/icons/edit.svg';
+
+import { Box, BoxProps, Image, Input } from '.';
+import { Body } from './Typography';
+
+export const EditableText = ({
+  value,
+  onChange,
+  placeholder,
+  ...props
+}: { value: string; placeholder?: string; onChange(value: string): void } & Omit<
+  BoxProps,
+  'onChange'
+>) => {
+  const [editMode, setEditMode] = useState(false);
+  const [editValue, setEditValue] = useState('');
+
+  const handleKeyDown = ({ key }: { key: string }) => {
+    if (key === 'Escape') {
+      handleCancel();
+    } else if (key === 'Enter') {
+      handleSave();
+    }
+  };
+
+  useEffect(() => {
+    setEditValue(value);
+  }, [value]);
+
+  const handleEdit = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    setEditMode(true);
+  };
+
+  const handleCancel = () => {
+    setEditMode(false);
+    setEditValue(value);
+  };
+
+  const handleSave = () => {
+    onChange(editValue);
+    setEditMode(false);
+  };
+
+  const hasValue = value !== undefined;
+
+  return (
+    <Box variant="rowAlign" height="100%" {...props}>
+      {editMode ? (
+        <Input
```

Comment:
This input is being changed from uncontrolled to controlled when writing something in it. I´m not sure why, but this should be avoided if possible.